blivet/populator/populator.py

```diff
@@ -262,7 +263,20 @@ def handle_device(self, info, update_orig_fmt=False):
             helper_class = self._get_device_helper(info)
 
         if helper_class is not None:
-            device = helper_class(self, info).run()
+            try:
+                device = helper_class(self, info).run()
+            except DeviceTreeError as e:
```

Comment:
What could be really important here is that every `run()` method should make sure to gather as much information as possible before throwing the exception. Otherwise the information about the problematic device may be very incomplete.